### PR TITLE
NIFI-10451 Updating QuestDB based status history rollover mechanism to fit the new major version

### PR DESCRIPTION
**Pull request by simonbence**
[NIFI-10451](https://issues.apache.org/jira/browse/NIFI-10451)

# Summary

[NIFI-00000](https://issues.apache.org/jira/browse/NIFI-00000)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/history/EmbeddedQuestDbRolloverHandler.java [hunk 1]
```diff
@@ -91,7 +95,8 @@ private void rolloverTable(final CharSequence tableName) {
 
     private void deletePartition(final CharSequence tableName, final String partition) {
         try (final SqlCompiler compiler = dbContext.getCompiler()) {
-            compiler.compile(String.format(DELETION_QUERY, new Object[]{tableName, partition}), dbContext.getSqlExecutionContext());
+            final CompiledQuery compile = compiler.compile(String.format(DELETION_QUERY, new Object[]{tableName, partition}), dbContext.getSqlExecutionContext());
+            compile.execute(new SCSequence(new TimeoutBlockingWaitStrategy(5, TimeUnit.SECONDS)));
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/history/EmbeddedQuestDbRolloverHandler.java:99]: Prior to this change, was the delete query not being executed?

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the improvement @simonbence! This looks straightforward, just noted a minor comment and implementation question.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/history/EmbeddedQuestDbRolloverHandler.java (hunk 1 above)
Comment (simonbence) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/status/history/EmbeddedQuestDbRolloverHandler.java:99]: Yes it was. In this particular case, 4.x did not need a call on the execute. At the time `EmbeddedQuestDbRolloverHandlerTest` was not ignored and they showed that the deletion had affect on the database. Also manual testing had the same results.

**Review by exceptionfactory — APPROVED** [review]
Thanks for the feedback and adjustments @simonbence, the latest version looks good! +1 merging

**Pull request closed without merging**